slang-gfx.h

```diff
@@ -1183,6 +1178,266 @@ struct WindowHandle
+};
+#define SLANG_UUID_IComputeCommandEncoder                                              \
+    {                                                                                  \
+        0x65400452, 0xc877, 0x478f, { 0x91, 0x7d, 0x48, 0xd5, 0x41, 0x6f, 0x39, 0xab } \
+    }
+
+class IResourceCommandEncoder : public ICommandEncoder
+{
+public:
+    virtual SLANG_NO_THROW void SLANG_MCALL copyBuffer(
+        IBufferResource* dst,
+        size_t dstOffset,
+        IBufferResource* src,
+        size_t srcOffset,
+        size_t size) = 0;
+    virtual SLANG_NO_THROW void SLANG_MCALL
+        uploadBufferData(IBufferResource* dst, size_t offset, size_t size, void* data) = 0;
+};
+#define SLANG_UUID_IResourceCommandEncoder                                             \
+    {                                                                                  \
+        0x5fe87643, 0x7ad7, 0x4177, { 0x8b, 0xd1, 0xd7, 0x84, 0xad, 0xcf, 0x3d, 0xce } \
+    }
+
+class ICommandBuffer : public ISlangUnknown
+{
+public:
+    // Only one encoder may be open at a time. User must call `ICommandEncoder::endEncoding`
+    // before calling other `encode*Commands` methods.
+    // Once `endEncoding` is called, the `ICommandEncoder` object becomes obsolete and is
+    // invalid for further use. To continue recording, the user must request a new encoder
+    // object by calling one of the `encode*Commands` methods again.
+    virtual SLANG_NO_THROW void SLANG_MCALL encodeRenderCommands(
+        IRenderPassLayout* renderPass,
+        IFramebuffer* framebuffer,
+        IRenderCommandEncoder** outEncoder) = 0;
+    ComPtr<IRenderCommandEncoder>
+        encodeRenderCommands(IRenderPassLayout* renderPass, IFramebuffer* framebuffer)
+    {
+        ComPtr<IRenderCommandEncoder> result;
+        encodeRenderCommands(renderPass, framebuffer, result.writeRef());
+        return result;
+    }
+
+    virtual SLANG_NO_THROW void SLANG_MCALL
+        encodeComputeCommands(IComputeCommandEncoder** outEncoder) = 0;
+    ComPtr<IComputeCommandEncoder> encodeComputeCommands()
+    {
+        ComPtr<IComputeCommandEncoder> result;
+        encodeComputeCommands(result.writeRef());
+        return result;
+    }
+
+    virtual SLANG_NO_THROW void SLANG_MCALL
+        encodeResourceCommands(IResourceCommandEncoder** outEncoder) = 0;
+    ComPtr<IResourceCommandEncoder> encodeResourceCommands()
+    {
+        ComPtr<IResourceCommandEncoder> result;
+        encodeResourceCommands(result.writeRef());
+        return result;
+    }
+
+    virtual SLANG_NO_THROW void SLANG_MCALL close() = 0;
+};
+#define SLANG_UUID_ICommandBuffer                                                      \
+    {                                                                                  \
+        0x5d56063f, 0x91d4, 0x4723, { 0xa7, 0xa7, 0x7a, 0x15, 0xaf, 0x93, 0xeb, 0x48 } \
+    }
+
+class ICommandQueue : public ISlangUnknown
+{
+public:
+    enum class QueueType
+    {
+        Graphics
```

Comment:
Any reason not to include the other cases here? I guess it is more implementation work to support all the case and our apps only ever want graphics for now.

Reply:
Yeah, I simply don't have time to implement all the other cases yet.